modules/kafka/src/main/scala/higherkindness/mu/format/AvroWithSchema.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu.format
+
+import java.io.ByteArrayOutputStream
+import com.sksamuel.avro4s._
+
+object AvroWithSchema {
+  implicit def encoder[T: SchemaFor: ToRecord]: Encoder[T] = new Encoder[T] {
+    override def encode(t: T): Array[Byte] = {
+      val bOut = new ByteArrayOutputStream()
+      val out  = AvroOutputStream.data[T](bOut)
+      out.write(t)
+      out.close()
+      bOut.toByteArray
+    }
+  }
+
+  implicit def decoder[T: SchemaFor: FromRecord]: Decoder[T] = new Decoder[T] {
+    override def decode(bytes: Array[Byte]): T = {
+      val in = AvroInputStream.data[T](bytes)
+      in.close()
+      in.iterator.toSet.head
```

Comment:
Same that above. In this case, it seems we're not closing the stream in the internal
https://github.com/higherkindness/mu-scala/blob/136f81ea0fb2b2d1a31089895b0724fccd7a3018/modules/internal/src/main/scala/higherkindness/mu/rpc/internal/encoders/avro.scala#L53-L54

Reply:
In this file, I don't think the `in.close()` is doing anything, because the AvroInputStream is wrapping a `SeekableByteArrayInput`, whose `close()` method is a no-op.

In the file linked by @fedefernandez, I think we're doing the right thing? The gRPC Javadoc doesn't say anything about whether we should close the stream, but the standard Java idiom is that you _don't_ close a resource you were passed as an argument. Closing the stream is the responsibility of its creator.